core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -144,49 +152,26 @@ public void build(ImageConfiguration imageConfig) throws Fabric8ServiceException
-        // Adding S2I artifacts such as environment variables in S2I mode
-        ArchiverCustomizer customizer = getS2ICustomizer(imageConfig);
-
-        try {
-            // Create tar file with Docker archive
-            File dockerTar;
-            if (customizer != null) {
-                dockerTar = dockerServiceHub.getArchiveService().createDockerBuildArchive(imageConfig, config.getDockerMojoParameters(), customizer);
-            } else {
-                dockerTar = dockerServiceHub.getArchiveService().createDockerBuildArchive(imageConfig, config.getDockerMojoParameters());
-            }
-            return dockerTar;
-        } catch (MojoExecutionException e) {
-            throw new Fabric8ServiceException("Unable to create the build archive", e);
+    private File createEmptyTargetDir() throws MojoExecutionException {
+        File targetDir = new File(config.getBuildDirectory(), "s2i");
+        if (!targetDir.exists() && !targetDir.mkdirs()) {
+            throw new MojoExecutionException("Can't create build working directory " + targetDir);
         }
+        return targetDir;
     }
 
-    private ArchiverCustomizer getS2ICustomizer(ImageConfiguration imageConfiguration) throws Fabric8ServiceException {
-        try {
-            if (imageConfiguration.getBuildConfiguration() != null && imageConfiguration.getBuildConfiguration().getEnv() != null) {
-                String fileName = IoUtil.sanitizeFileName("s2i-env-" + imageConfiguration.getName());
-                final File environmentFile = new File(config.getBuildDirectory(), fileName);
+    private File createTar(File targetDir) throws IOException, MojoExecutionException {
+        TarArchiver tarArchiver = new TarArchiver();
+        tarArchiver.addDirectory(targetDir);
 
-                try (PrintWriter out = new PrintWriter(new FileWriter(environmentFile))) {
-                    for (Map.Entry<String, String> e : imageConfiguration.getBuildConfiguration().getEnv().entrySet()) {
-                        out.println(e.getKey() + "=" + e.getValue());
-                    }
-                }
+        // Delete previously existing tar files, if any
+        FileUtil.deleteFilesInDirectory(targetDir, (file, s) -> { return s.endsWith(".tar"); });
 
-                return new ArchiverCustomizer() {
-                    @Override
-                    public TarArchiver customize(TarArchiver tarArchiver) throws IOException {
-                        tarArchiver.addFile(environmentFile, ".s2i/environment");
-                        return tarArchiver;
-                    }
-                };
-            } else {
-                return null;
-            }
-        } catch (IOException e) {
-            throw new Fabric8ServiceException("Unable to add environment variables to the S2I build archive", e);
-        }
+        File tarFile = File.createTempFile(config.getArtifactId().toString(), ".tar");
```

Comment:
Not sure if I understand this: `tarFile` is a temporary tar filename but the tar archive itself is created with the `destFile` as name. The return value is then a the temporary name (which doesn't point to any tar archive if I understood the code right). Could you please check this again ?